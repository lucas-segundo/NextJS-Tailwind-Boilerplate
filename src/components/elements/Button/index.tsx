import { ButtonHTMLAttributes } from 'react'

export type ButtonProps = {
  color?: ButtonColorEnum
  isPill?: boolean
  children: React.ReactNode
} & ButtonHTMLAttributes<HTMLButtonElement>

export enum ButtonColorEnum {
  BLUE = 'bg-blue-500',
  YELLOW = 'bg-yellow-400'
}

const Button = ({
  color = ButtonColorEnum.BLUE,
  isPill,
  children,
  ...props
}: ButtonProps) => (
  <button
    {...props}
    type="button"
    className={`px-4 py-2 ${props.className} ${color} ${
      isPill && 'rounded-full'
    } text-white font-bold`}
  >
    {children}
  </button>
)

export default Button
