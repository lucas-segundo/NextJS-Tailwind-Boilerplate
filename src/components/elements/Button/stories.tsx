import { Story, Meta } from '@storybook/react'
import Button, { ButtonProps } from '.'

export default {
  title: 'Elements/Button',
  component: Button
} as Meta

export const Default: Story<ButtonProps> = (args) => <Button {...args} />

Default.args = {
  isPill: false,
  children: 'Button'
}
