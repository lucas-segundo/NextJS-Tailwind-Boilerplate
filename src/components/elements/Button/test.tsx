import { render, screen } from '@testing-library/react'

import Button, { ButtonColorEnum } from '.'

describe('<Button />', () => {
  it('should render correctly', () => {
    const { container } = render(<Button>Button</Button>)

    expect(screen.getByRole('button', { name: /Button/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render like pill', () => {
    const { container } = render(<Button isPill>Test Button</Button>)

    expect(screen.getByRole('button', { name: /Test Button/i })).toHaveClass(
      'rounded-full'
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render yellow', () => {
    const { container } = render(
      <Button color={ButtonColorEnum.YELLOW}>Test Button</Button>
    )

    expect(screen.getByRole('button', { name: /Test Button/i })).toHaveClass(
      ButtonColorEnum.YELLOW
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
