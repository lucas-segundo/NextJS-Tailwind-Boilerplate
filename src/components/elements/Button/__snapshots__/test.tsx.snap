// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly 1`] = `
<button
  class="px-4 py-2 undefined bg-blue-500 undefined text-white font-bold"
  type="button"
>
  Button
</button>
`;

exports[`<Button /> should render like pill 1`] = `
<button
  class="px-4 py-2 undefined bg-blue-500 rounded-full text-white font-bold"
  type="button"
>
  Test Button
</button>
`;

exports[`<Button /> should render yellow 1`] = `
<button
  class="px-4 py-2 undefined bg-yellow-400 undefined text-white font-bold"
  type="button"
>
  Test Button
</button>
`;
